orchestrator/db/filters/process.py

```diff
@@ -11,148 +11,101 @@
-
-
-BASE_CAMEL = {to_camel(key): generic_is_like_filter(value) for key, value in inspect(ProcessTable).columns.items()}
-BASE_SNAKE = {key: generic_is_like_filter(value) for key, value in inspect(ProcessTable).columns.items()}
-
-PROCESS_FILTER_FUNCTIONS_BY_COLUMN: dict[str, Callable[[QueryType, str], QueryType]] = (
-    BASE_CAMEL
-    | BASE_SNAKE
-    | {
-        "isTask": generic_bool_filter(ProcessTable.is_task),
-        "is_task": generic_bool_filter(ProcessTable.is_task),
-        "assignee": generic_values_in_column_filter(ProcessTable.assignee),
-        "lastStatus": generic_values_in_column_filter(ProcessTable.last_status),
-        "product": product_filter,
-        "productTag": tag_filter,
-        "tag": tag_filter,
-        "subscriptions": subscriptions_filter,
-        "subscriptionId": subscription_id_filter,
-        "subscription_id": subscription_id_filter,
-        "target": target_filter,
-        "workflow_name": workflow_name_filter,
-        "workflowName": workflow_name_filter,
-        "customer": customer_filter,
-    }
-)
 
+    return ProcessTable.process_id.in_(process_workflow)
 
-def make_product_clause(filter_generator: WhereCondGenerator) -> WhereCondGenerator:
-    """The passed filter_generator takes a Node and returns a where clause acting on a ProductTable column."""
 
-    def product_clause(node: Node) -> BinaryExpression:
-        process_subscriptions = (
-            select(ProcessSubscriptionTable.process_id)
-            .join(SubscriptionTable)
-            .join(ProductTable)
-            .where(filter_generator(node))
-            .subquery()
-        )
-        return ProcessTable.process_id.in_(process_subscriptions)
-
-    return product_clause
+def workflow_targets_clause(node: Node) -> BinaryExpression:
+    process_workflows = (
+        select(ProcessTable.process_id)
+        .join(WorkflowTable)
+        .where(WorkflowTable.deleted_at.is_(None), WorkflowTable.target == node_to_str_val(node))
+        .subquery()
+    )
+    return ProcessTable.process_id.in_(process_workflows)
 
 
 PROCESS_TABLE_COLUMN_CLAUSES = default_inferred_column_clauses(ProcessTable) | {
+    "customer": customer_clause,
     "product": make_product_clause(inferred_filter(ProductTable.name)),
     "product_description": make_product_clause(inferred_filter(ProductTable.description)),
+    "subscription_id": make_subscription_id_clause(inferred_filter(SubscriptionTable.subscription_id)),
     "tag": make_product_clause(filter_exact(ProductTable.tag)),
+    "target": workflow_targets_clause,
+    "workflow_name": workflow_name_clause,
 }
 
 
-process_filter_fields = list(PROCESS_FILTER_FUNCTIONS_BY_COLUMN.keys())
-filter_processes = generic_filter(PROCESS_FILTER_FUNCTIONS_BY_COLUMN)
+process_filter_fields = create_memoized_field_list(PROCESS_TABLE_COLUMN_CLAUSES)
+filter_processes = generic_filter_from_clauses(PROCESS_TABLE_COLUMN_CLAUSES)
```

Comment:
Just to check whether I'm on the right page:
* These definitions  no longer contain camelCase keys but only snake_case keys
* The search query filters can be any of the 2: it is just normalized to snake_case
* In the error response to a query (in case of invalid filters) we send back the keys in camelCase because for GQL that makes the most sense

Reply:
Correct. Everything concertning the FE/JSON is camelCase, but in the inner python code we work with snake_case